### ISSUE TITLE
comma instead of semicolon with collapsing and locator on second citation

### ISSUE THREAD

**Issue by odkr** [jgm/citeproc#39]
The Chicago Manua of Style requires that multiple sources are separated by semicolons, rather than commas, if pages are given for at least one of them.

The way to achieve this behaviour used to be to prefix the semicolon that separates multiple sources in the Markdown source code by `\;`. Pandoc then omitted the comma. It no longer does.

```sh
pandoc -C -t plain <<EOF
---
references:
    - type: book
      id: one
      author: Jane Doe
      issued:
          date-parts:
              - - 2010
      title: The joy of citations
    - type: book
      id: two
      author: Jane Doe
      issued:
          date-parts:
              - - 2020
      title: Citations in times of COVID-19
---

P [@one\;; @two]!
EOF
```

This outputs:

> P (Jane Doe 2010;, 2020)!
> [...]

The output I expected is:

> P (Jane Doe 2010; 2020)!
> [...]

I am using pandoc v2.11.2 on macOS 11.1.

If somebody else stumbles upon this thread, this Lua filter replaces ";," by ";" in citations:

```Lua
function Cite (cite)
    for i = 1, #cite.c do
        local els = cite.c[i]
        for j = 1, #els do
            local el = els[j]
            if el.t == 'Str' then el.c = el.c:gsub(';,', ';') end
        end
    end
    return cite
end
```

Thanks a for looking into this.

**Comment by jgm** [comment]
Some useful info here:
https://github.com/citation-style-language/test-suite/issues/36
This is a bit of a mess if I recall.

**Comment by jgm** [comment]
Relevant code in citeproc is `collapseGroup`. This needs some rethinking.

**Comment by odkr** [comment]
Wow. Thanks a lot for your effort!

**Comment by jgm** [comment]
Closed by d0e54865837a44ad0109867b85a4b88bb427c9c1
